model/src/lib.rs

```diff
@@ -11,40 +12,33 @@ use lazy_static::lazy_static;
 use primitive_types::{H160, H256};
 use serde::{de, Deserialize, Serialize};
 use std::fmt;
-use web3::signing;
+use web3::signing::{self};
 use web3::signing::{Key, SecretKeyRef};
 use web3::types::Recovery;
 
+#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "lowercase")]
+pub enum SigningScheme {
+    EIP712,
+}
+
 #[derive(Eq, PartialEq, Clone, Copy, Debug, Default, Hash)]
 pub struct Signature {
     pub r: H256,
     pub s: H256,
     pub v: u8,
 }
 
-pub trait EIP712Signing {
```

Comment:
I am removing this trait because with the new signing schemes I found it hard to wrap my head around implementing them and keeping this trait consistent. In the produce of adding support for other signing schemes we will see how we can generalize again to go back to having a trait.